flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/formats/raw/RawFormatFactory.java

```diff
@@ -0,0 +1,186 @@
+		.withDescription("Defines the endianness of the bytes of the numeric value.");
+
+	public static final ConfigOption<String> CHARSET = ConfigOptions
+		.key("charset")
+		.stringType()
+		.defaultValue(StandardCharsets.UTF_8.displayName())
+		.withDescription("Defines the string charset.");
+
+	@Override
+	public String factoryIdentifier() {
+		return IDENTIFIER;
+	}
+
+	@Override
+	public Set<ConfigOption<?>> requiredOptions() {
+		return Collections.emptySet();
+	}
+
+	@Override
+	public Set<ConfigOption<?>> optionalOptions() {
+		Set<ConfigOption<?>> options = new HashSet<>();
+		options.add(ENDIANNESS);
+		options.add(CHARSET);
+		return options;
+	}
+
+	@Override
+	public DecodingFormat<DeserializationSchema<RowData>> createDecodingFormat(
+			Context context,
+			ReadableConfig formatOptions) {
+		FactoryUtil.validateFactoryOptions(this, formatOptions);
+		final String charsetName = validateAndGetCharsetName(formatOptions);
+		final boolean isBigEndian = isBigEndian(formatOptions);
+
+		return new DecodingFormat<DeserializationSchema<RowData>>() {
+			@Override
+			public DeserializationSchema<RowData> createRuntimeDecoder(
+					DynamicTableSource.Context context,
+					DataType producedDataType) {
+				final RowType physicalRowType = (RowType) producedDataType.getLogicalType();
+				final LogicalType fieldType = validateAndExtractSingleField(physicalRowType);
+				final TypeInformation<RowData> producedTypeInfo = context.createTypeInformation(producedDataType);
+				return new RawFormatDeserializationSchema(fieldType, producedTypeInfo, charsetName, isBigEndian);
+			}
+
+			@Override
+			public ChangelogMode getChangelogMode() {
+				return ChangelogMode.insertOnly();
+			}
+		};
+	}
+
+	@Override
+	public EncodingFormat<SerializationSchema<RowData>> createEncodingFormat(
+			Context context, ReadableConfig formatOptions) {
+		FactoryUtil.validateFactoryOptions(this, formatOptions);
+		final String charsetName = validateAndGetCharsetName(formatOptions);
+		final boolean isBigEndian = isBigEndian(formatOptions);
+
+		return new EncodingFormat<SerializationSchema<RowData>>() {
+			@Override
+			public SerializationSchema<RowData> createRuntimeEncoder(
+					DynamicTableSink.Context context,
+					DataType consumedDataType) {
+				final RowType physicalRowType = (RowType) consumedDataType.getLogicalType();
+				final LogicalType fieldType = validateAndExtractSingleField(physicalRowType);
+				return new RawFormatSerializationSchema(fieldType, charsetName, isBigEndian);
+			}
+
+			@Override
+			public ChangelogMode getChangelogMode() {
+				return ChangelogMode.insertOnly();
+			}
+		};
+	}
+
+	// ------------------------------------------------------------------------------------------
+
+	/**
+	 * Validates and extract the single field type from the given physical row schema.
+	 */
+	private static LogicalType validateAndExtractSingleField(RowType physicalRowType) {
+		if (physicalRowType.getFieldCount() != 1) {
```

Comment:
shall we perform real type validation here? e.g. TIMESTAMP or NULL would still pass